### DISCUSSION TITLE
Is it possible to change the default home/landing to my own controller /Home/Index ?

### DISCUSSION THREAD

**Discussion by orchardc** [OrchardCMS/OrchardCore#15862] [Q&A]
As I'd like to redirect the homepage to use my own controller /Home/Index

**Comment by Piedone** [comment]
You don't need to redirect anything. If you specify your controller action's route as the home route (I think just "/" should work), like with [attribute routing](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/orchard-core/Controllers/YourFirstOrchardCoreController.cs#L60) then that's what will come up when you open the homepage.

**Comment by orchardc** [comment]
Hi, this doesn't work:
            routes.MapAreaControllerRoute(
                name: "Home",
                areaName: "MyCode",
                pattern: "/",
                defaults: new { controller = "MyController", action = "Index" }
            );

**Comment by Piedone** [comment]
How exactly is it not working?

**Comment by orchardc** [comment]
It ignore this route, still directly go to the landing page.

**Comment by Piedone** [comment]
Let's try to make this work in a simplest way: Please create a controller in your module where the action just returns a string or something, and use attribute routing to hook it to the homepage. Enable the module, then try it out. If it doesn't work then open the action directly with the default route. Let us know if this works and post your code.

**Comment by Skrypt** [comment]
You can change the homepage route by changing it in the database.

You will do this SQL Query to find the proper record in the DB : 

```SQL

SELECT * FROM Document WHERE Type = 'OrchardCore.Settings.SiteSettings, OrchardCore.Settings'

```

And then change this part of the JSON data with what you need : 

```JSON

{
  "HomeRoute": {
    "area": "OrchardCore.Contents",
    "controller": "Item",
    "action": "Display",
    "contentItemId": "12121212121212"
  }
}

```

Let's add this to the documentation please as this question is often asked. @agriffard

**Comment by agriffard** [comment]
@Skrypt I tried to create a Tools module with an admin page to set the HomeRoute.

The homeRoute is correctly updated but, it still needs to update the current one to SetHomepage = false and if there is a contentItemId, update the content that has the new contentItemId with SetHomepage = true.

If you want to take a look:
https://github.com/agriffard/Tools.OrchardCore/blob/41918a67495dc4866b74a5c49720407412cee00c/src/Tools/Services/ToolsService.cs#L31

**Comment by jtkech** [comment]
@orchardc 

You just need to disable the `HomeRoute` feature, no feature really depends on it, `Autoroute` just provides a way to define the home route settings that are then used by the `HomeRoute` feature

So if you disable the  `HomeRoute` feature you can then define an home page in a regular mvc way, as @Piedone suggested, with a route attribute `[Route("")]` or by using a `MapAreaControllerRoute()`

    public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, ...)
    {
        routes.MapAreaControllerRoute(
            name: "Html.Demo.Test",
            areaName: "OrchardCore.Html", <= needs to be the module assembly name
            pattern: "",
            defaults: new { controller = "Demo", action = "Test" }
        );
    }

**Comment by Piedone** [comment]
Just for those stumbling on this, looking for a solution to override other routes than the home route: with the `Order` property of a `Startup` class, you can make it run before the `Startup` of the module whose route you want to override. E.g.:

```c#
public class Startup : StartupBase
{
    public override int Order => -100;

    public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
    {
        var userOptions = serviceProvider.GetRequiredService<IOptions<UserOptions>>().Value;

        routes.MapAreaControllerRoute(
            name: "UsersLogOff",
            areaName: "MyModule",
            pattern: userOptions.LogoffPath,
            defaults: new
            {
                controller = typeof(MyAccountController).ControllerName(),
                action = nameof(MyAccountController.LogOff),
            }
        );
    }
}
```
